cmake/thirdparty/FindCppcheck.cmake

```diff
@@ -50,14 +50,19 @@
 # CPPCHECK_EXECUTABLE
 ################################################################################
 
-find_program(CPPCHECK_EXECUTABLE
-             NAMES cppcheck
-             DOC "Path to cppcheck executable")
+if (CPPCHECK_EXECUTABLE)
```

Comment:
@white238 -- It looks like you are using the same logic for all these.  Can you convert this to a (private) macro to ensure consistency?

Reply:
I tried to make this into a nice general macro and it went poorly (too many things to guard against and cover).  I could definitely make it a private macro that only covers our one case.

Reply:
A single purpose private macro sounds good.  (Or leave it for the future if you think it would be too messy).